src/main/java/com/hotels/beeju/core/HiveMetaStoreCore.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.core;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+
+public class HiveMetaStoreCore {
+
+  private HiveMetaStoreClient client;
+  private final BeejuCore beejuCore;
+
+  public HiveMetaStoreCore(BeejuCore beejuCore) {
+    this.beejuCore = beejuCore;
+  }
+
+  public void before() throws InterruptedException, ExecutionException {
```

Comment:
I think in the juni4 rule context the names `before` and `after` but not anymore, maybe rename them in `initialise` and  `shutdown`??.  Same for the other core classes that have this naming.